### PR TITLE
Added a check to detect if AE2 and the API are loaded from the same source

### PR DESCRIPTION
**Pull request by yueh**
In favour of #1167.

I am able to reproduce it reliably with [this ThaumicTinkerer build](http://jenkins.nekokittygames.com:8080/job/Thaumic%20Tinkerer/439/)

### CODE REVIEW

File: src/main/java/appeng/core/crash/ApiConflictCrashEnhancement.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import javax.annotation.Nonnull;
+
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+
+
+public class ApiConflictCrashEnhancement extends BaseCrashEnhancement
+{
+
+	private static final Function<String, String> TRANSFORM_CONFLICTING_MOD = new Function<String, String>(){
```
Comment (orod-org) [inline src/main/java/appeng/core/crash/ApiConflictCrashEnhancement.java]: MINOR Move this left curly brace to the beginning of next line of code. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

File: src/main/java/appeng/core/features/AEFeature.java [hunk 2]
```diff
@@ -122,6 +122,7 @@ public boolean isVisible()
 	CraftingLog( Constants.CATEGORY_MISC, false ),
 	LightDetector( Constants.CATEGORY_MISC ),
 	DebugLogging( Constants.CATEGORY_MISC, false ),
+	ApiConflictCheck( Constants.CATEGORY_MISC, true ),
```
Comment (orod-org) [inline src/main/java/appeng/core/features/AEFeature.java]: MINOR Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00115)

File: src/main/java/appeng/core/AppEng.java [hunk 3]
```diff
@@ -241,4 +254,48 @@ private void serverStarting( final FMLServerStartingEvent evt )
...
+	private void checkForApiConflicts()
+	{
+		final ModAPIManager apiManager = ModAPIManager.INSTANCE;
+		if( apiManager.hasAPI( API_ID ) )
+		{
+			final ModContainer aeContainer = FMLCommonHandler.instance().findContainerFor( MOD_ID );
+			ModContainer aeApiContainer = null;
+
+			final Iterable<? extends ModContainer> apiList = apiManager.getAPIList();
+			for( ModContainer container : apiList )
```
Comment (orod-org) [inline src/main/java/appeng/core/AppEng.java]: MINOR Local variable 'container' could be declared final [rule](http://nemo.sonarqube.org/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)

**Comment by orod-org** [comment]
SonarQube analysis reported 3 issues:
* MINOR 3 minor

Watch the comments in this conversation to review them.

**Comment by yueh** [comment]
Small update. The exception can now be disabled inside the config and it will now log its state and potentially mod shipping the API to any crashlog similar to the AE2 version.

**Comment by orod-org** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Pull request closed without merging**